### DISCUSSION TITLE
Excluding unused locales from webpack bundle

### DISCUSSION THREAD

**Discussion by pawel-marciniak** [js-fns/date-fns#1797] [General] [answered]
Hello,

Has anyone got working webpack configuration (ContextReplacementPlugin or another way) to exclude unwanted localization files from bundle? I've tried the way from documentation and different variations of regexp and nothing works so far.

**Comment by taco-tues-on-a-fri** [comment, marked as answer]
@psimk  's posted a [link to a repo](https://github.com/psimk/date-fns-webpack-context-replacement-plugin-example/blob/master/webpack.working.config.js) of his in [this thread](https://github.com/date-fns/date-fns/issues/1697) with a working example that helped me get around the regex issue with webpack throwing errors from the `.js.flow` & `.md` files that were also getting included in the dynamic import of locales.

He opened a [PR](https://github.com/date-fns/date-fns/pull/1827) to make the change in the documentation.

```
new webpack.ContextReplacementPlugin(
  /date\-fns[\/\\]/,
  new RegExp(`[/\\\\\](${supportedLocales.join('|')})[/\\\\\]index\.js$`)
 )
```

Hopefully this helps anyone looking for a solution.
